nbviewer/providers/gist/handlers.py

```diff
@@ -111,94 +129,139 @@ def get(self, user, gist_id, filename=''):
+            return
+        
+        yield self.deliver_notebook(user, gist_id, filename, gist, file, content)
+
+    @gen.coroutine
+    def get_notebook_data(self, gist_id, filename, many_files_gist, file):
+        if (file['type'] or '').startswith('image/'):
+            app_log.debug("Fetching raw image (%s) %s/%s: %s", file['type'], gist_id, filename, file['raw_url'])
+            response = yield self.fetch(file['raw_url'])
+            # use raw bytes for images:
+            content = response.body
+        elif file['truncated']:
+            app_log.debug("Gist %s/%s truncated, fetching %s", gist_id, filename, file['raw_url'])
+            response = yield self.fetch(file['raw_url'])
+            content = response_text(response, encoding='utf-8')
         else:
-            entries = []
-            ipynbs = []
-            others = []
-
-            for file in files.values():
-                e = {}
-                e['name'] = file['filename']
-                if file['filename'].endswith('.ipynb'):
-                    e['url'] = quote('/%s/%s' % (gist_id, file['filename']))
-                    e['class'] = 'fa-book'
-                    ipynbs.append(e)
-                else:
-                    provider_url = u"https://gist.github.com/{user}/{gist_id}#file-{clean_name}".format(
-                        user=user,
-                        gist_id=gist_id,
-                        clean_name=clean_filename(file['filename']),
-                    )
-                    e['url'] = provider_url
-                    e['class'] = 'fa-share'
-                    others.append(e)
-
-            entries.extend(ipynbs)
-            entries.extend(others)
-
-            # Enable a binder navbar icon if a binder base URL is configured
-            executor_url = self.BINDER_TMPL.format(
-                binder_base_url=self.binder_base_url,
-                user=user.rstrip('/'),
-                gist_id=gist_id
-            ) if self.binder_base_url else None
-
-            html = self.render_template(
-                'treelist.html',
-                entries=entries,
-                tree_type='gist',
-                tree_label='gists',
-                user=user.rstrip('/'),
-                provider_url=gist['html_url'],
-                executor_url=executor_url,
-                **self.PROVIDER_CTX
-            )
-            yield self.cache_and_finish(html)
+            content = file['content']
+
+        if many_files_gist and not filename.endswith('.ipynb'):
+            self.set_header('Content-Type', file.get('type') or 'text/plain')
+            # cannot redirect because of X-Frame-Content
+            self.finish(content)
+            return
+
+        else:
+            return content
+
+    @gen.coroutine
+    def deliver_notebook(self, user, gist_id, filename, gist, file, content):
+        """
+        provider_url: str, optional
+            URL to the notebook document upstream at the provider (e.g., GitHub)
```

Comment:
We should remove `provider_url` from the docstring since it's not a parameter to this function and optionally document the actual parameters.